### ISSUE TITLE
screen blinks black when using gdb-imagewatch

### ISSUE THREAD

**Issue by kameamea** [csantosbh/gdb-imagewatch#12]
This is a really strange behaviour which makes gdb-imagewatch unusable on my computer:

If I use it, or just issue "python3 gdb-imagewatch.py --test" the gdb-imagewatch window appears, but after less than a second the whole screen gets black. A few seconds later the screen reappears for a fraction of a second and this repeats forever until I kill gdb-imagewatch. It seems like the monitor has synchronisation problems (which it never has with any other application), it's a flickering like when turning on the PC or switching to the Ctrl-Alt-F1 console . And: I can use gdb-imagewatch with no problems on the same host+monitor if I issue it from a different host with ssh and DISPLAY set to the machine where it fails.

So to summarize:
HostA: ubuntu 16.04.2, intel skylake-processor with integrated GPU and 4K-Monitor attached
-> gdb-imagewatch causes monitor synchronisation problems

HostB: ubuntu 16.04.2 
-> no problem when using
HostA% ssh HostB
HostB% export DISPLAY=HostA:0
HostB% python3 gdb-imagewatch.py --test

Do you have any idea what might cause this behaviour?

**Comment by csantosbh** [comment]
Hm this is strange. If I recall correctly, you were able to use gdb-imagewatch before on this same machine, right? If so, there must be something wrong with my recent modifications.

Anyways, I just reproduced a somewhat similar issue on my machine while performing a seemingly innocent refactoring: The screen is flickering very rapidly, as if the renderer were alternating between black frames and the actual buffer that should be rendered. I'll take a look at the recent changes to see what could cause this problem. Thanks again for reporting!

**Comment by csantosbh** [comment]
Update: Turns out the problem was not my refactor. As I realized, even very old versions of the repository were presenting the flickering issue. Then I remembered I was playing with a more recent version of the mesa driver (I installed version 17.2.0-dev from ppa:oibaf/graphics-drivers to play a bit with OpenCL). Downgrading back to the stable version 17.0.7 solved the issue for me. If you are unsure about your mesa version, you can check it out by running:
```
glxinfo | grep "OpenGL version"
```

By the way, if downgrading is not an option for you, I realized that the flickering could be avoided by reducing the rendering framerate. That can be done by increasing the `16` in `update_timer_.start(16);` (mainwindow.cpp, inside MainWindow::show). In my tests, using 48 (which is equivalent to 20fps) did the trick for me.

**Comment by kameamea** [comment]
my mesa driver is on 12.0.6. But increasing the update timer to 48 indeed helped! Probably some part of my graphic driver chain was just overwhelmed by 60fps@4K. OTOH: why would you want 60fps for this kind of application with static images. 20fps should be more than enough as default.

**Comment by csantosbh** [comment]
I agree that 60fps may be quite too high for most use cases. There was no particular reason for this value other than wanting the user experience to be as smooth as possible, but the least I should do is make this an option that the user can easily modify the framerate.

Also, considering that you are using a stable version of mesa, I'll reopen the issue to work on a proper solution. I might be running into a case where multiple draws are being requested on top of each other, without waiting for the previous ones to finish.
